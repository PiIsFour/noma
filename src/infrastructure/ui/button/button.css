.button {
	display: inline-block;
	border: 0;
	border-radius: .5em;
	color: snow;
	background-color: transparent;
	box-shadow: snow 0px 0px 0px 1px inset;
	font-size: calc((10px + 2vmin) / 1.5);
	padding: 1em 3em;
	cursor: pointer;
	display: inline-block;
	line-height: 1;
}
