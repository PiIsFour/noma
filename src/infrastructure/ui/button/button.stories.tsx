import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import { Button } from './button'

export default {
	title: 'Button',
	component: Button,
} as ComponentMeta<typeof Button>

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />

export const Standard = Template.bind({})
Standard.args = {
	label: 'Click Me',
}
