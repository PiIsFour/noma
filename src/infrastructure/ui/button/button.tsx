import React from 'react'
import './button.css'

interface Props {
	label: string;
	onClick?: () => void;
}

export const Button = ({
	label,
	onClick,
}: Props): JSX.Element => {
	return <button
		type="button"
		className="button"
		onClick={onClick}
	>
		{label}
	</button>
}
