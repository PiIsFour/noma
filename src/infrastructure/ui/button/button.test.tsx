import React from 'react'
import { shallow } from 'enzyme'

import { Button } from './button'

describe('button', () => {
	it('matches snapshot', () => {
		const wrapper = shallow(<Button label="test button" />)
		expect(wrapper).toMatchSnapshot()
	})
})
