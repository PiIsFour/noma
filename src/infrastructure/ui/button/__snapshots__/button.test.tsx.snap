// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button matches snapshot 1`] = `
<button
  className="button"
  type="button"
>
  test button
</button>
`;
